### PR TITLE
[PXM-39] Added internal randomization of traits

### PR DESCRIPTION
**Pull request by GabrielCartier**

### CODE REVIEW

File: contracts/Genesis.sol [hunk 1]
```diff
@@ -226,25 +254,85 @@ contract Genesis is ERC721Pausable, VRFConsumerBase, Ownable {
...
-     * @param sender sender of the request 
-     * @param tokenId id of the token
-     * @return requestId id of the request sent to Chainlink
+     * Mint reserved gods
+     * @param to address to send the god to
+     * @param count number of gods to transfer
+     */
+    function mintReservedGods(address to, uint256 count) public onlyOwner {
+        require(
+            reservedGodsSupply() >= count + reservedGodsTransfered.current(),
```
Comment (GabrielCartier) [inline contracts/Genesis.sol:284]: this used to be `require(reservedGodsSupply() >= count, "Not enough reserved gods left");`, pretty sure that was not valid right?

**Comment by GabrielCartier** [comment]
Sorry, lots of prettified code.

File: .prettierrc [hunk 2]
```diff
@@ -3,5 +3,18 @@
...
+      "options": {
+        "printWidth": 80,
+        "tabWidth": 4,
+        "useTabs": false,
+        "singleQuote": false,
+        "bracketSpacing": false,
+        "explicitTypes": "always"
+      }
+    }
+  ]
```
Comment (0xjohnnycagewins) [inline .prettierrc:19]: we should not have to specify this as the plugin does exactly this

File: contracts/Genesis.sol [hunk 3]
```diff
@@ -212,12 +231,21 @@ contract Genesis is ERC721Pausable, VRFConsumerBase, Ownable {
...
      * @param proof Proof to verify that the caller is allowed to mint
      */
-    function freeMint(uint count, uint256 maxMintCount, bytes32[] calldata proof) external whenNotPaused() onlyFreeMint(maxMintCount, proof) {
-        require((tokenCounter.current() + count) < MAX_SUPPLY, "Not enough supply");
-        uint mintCount = addressToMintCount[msg.sender] + addressToRequestCount[msg.sender] + count;
+    function freeMint(
+        uint256 count,
+        uint256 maxMintCount,
+        bytes32[] calldata proof
+    ) external whenNotPaused seedGenerated onlyFreeMint(maxMintCount, proof) {
```
Comment (0xjohnnycagewins) [inline contracts/Genesis.sol:238]: `seedGenerated` 👍🏼

File: contracts/Genesis.sol (hunk 1 above)
Comment (0xjohnnycagewins) [reply contracts/Genesis.sol:284]: yeah this will all change anyway....

File: contracts/Genesis.sol [hunk 4]
```diff
@@ -226,25 +254,85 @@ contract Genesis is ERC721Pausable, VRFConsumerBase, Ownable {
...
+     * @param randomNumber random number returned by Chainlink
+     */
+    function fulfillRandomness(bytes32 requestId, uint256 randomNumber)
+        internal
+        override
+    {
+        console.log("Request Id is");
+        console.logBytes32(requestId);
+        console.log("randomizationRequestId is");
+        console.logBytes32(randomizationRequestId);
```
Comment (0xjohnnycagewins) [inline contracts/Genesis.sol]: are we leaving this here and stripping them before deployment with the plugin?
if so we need to check out https://github.com/ItsNickBarry/hardhat-log-remover#readme

**Review by 0xjohnnycagewins — APPROVED** [review]

File: contracts/Genesis.sol (hunk 4 above)
Comment (GabrielCartier) [reply contracts/Genesis.sol]: shouldn't be there indeed, will remove, and good plugin, will add!

File: .prettierrc (hunk 2 above)
Comment (GabrielCartier) [reply .prettierrc:19]: can't remove it it seems...

**Pull request merged**